source/Handlebars/HtmlEncoder.cs

```diff
@@ -11,9 +11,36 @@ public string Encode(string text)
             if (text == null)
                 return String.Empty;
 
-            var sb = new StringBuilder(text.Length);
 
+            // Detect if we need to allocate a stringbuilder and new string
             for (var i = 0; i < text.Length; i++)
+            {
+                switch (text[i])
+                {
+                    case '"':
+                    case '&':
+                    case '<':
+                    case '>':
+                        return ReallyEncode(text, i);
+                    default:
+                        if (text[i] > 159)
+                        {
+                            return ReallyEncode(text, i);
+                        }
+                        else
+
+                            break;
+                }
+            }
+
+            return text;
+        }
+
+        private static string ReallyEncode(string text, int i)
+        {
+            var sb = new StringBuilder(text.Length + 5);
```

Comment:
@304NotModified Why are you adding 5 to the length? How does this help during allocation?